cms/static/js/spec/views/group_configuration_spec.js

```diff
@@ -626,4 +630,210 @@ define([
+                groupConfiguration = new GroupConfigurationModel({
+                    id: 0,
+                    name: 'Group Configuration',
+                    groups: groups
+                });
+            groupConfiguration.urlRoot = '/mock_url';
+            return groups;
+        };
+
+        renderView = function(groupNames) {
+            var view = new GroupList({collection: createGroups(groupNames || [])}).render();
+            appendSetFixtures(view.el);
+            return view;
+        };
+
+        saveOrCancel = function(view, options, groupIndex) {
+            if (options.save) {
+                view.$(scopedGroupSelector(groupIndex, saveButtonCss)).click();
+            } else if (options.cancel) {
+                view.$(scopedGroupSelector(groupIndex, cancelButtonCss)).click();
+            }
+        };
+
+        editNewGroup = function(view, options) {
+            var newGroupIndex;
+            if (view.collection.length === 0) {
+                view.$(newGroupCss).click();
+            } else {
+                view.$(addGroupCss).click();
+            }
+            newGroupIndex = view.collection.length - 1;
+            view.$(inputCss).val(options.newName);
+            verifyEditingGroup(view, true, newGroupIndex);
+            saveOrCancel(view, options, newGroupIndex);
+        };
+
+        editExistingGroup = function(view, options) {
+            var groupIndex = options.groupIndex || 0;
+            view.$(scopedGroupSelector(groupIndex, '.edit')).click();
+            view.$(scopedGroupSelector(groupIndex, inputCss)).val(options.newName);
+            saveOrCancel(view, options, groupIndex);
+        };
+
+        verifyEditingGroup = function(view, expectEditing, index) {
+            // Should prevent the user from opening more than one edit
+            // form at a time by removing the add button(s) when
+            // editing a group.
+            index = index || 0;
+            if (expectEditing) {
+                expect(view.$(scopedGroupSelector(index, '.group-configuration-edit'))).toExist();
+                expect(view.$(newGroupCss)).not.toExist();
+                expect(view.$(addGroupCss)).not.toExist();
+            } else {
+                expect(view.$('.group-configuration-edit')).not.toExist();
+                if (view.collection.length === 0) {
+                    expect(view.$(newGroupCss)).toExist();
+                    expect(view.$(addGroupCss)).not.toExist();
+                } else {
+                    expect(view.$(newGroupCss)).not.toExist();
+                    expect(view.$(addGroupCss)).toExist();
+                }
+            }
+        };
+
+        respondToSave = function(requests, view) {
+            expect(requests.length).toBe(1);
+            expect(requests[0].method).toBe('POST');
+            expect(requests[0].url).toBe('/mock_url/0');
+            AjaxHelpers.respondWithJson(requests, {
+                name: 'Group Configuration',
```

Comment:
Whenever content groups are saved, the enclosing cohort-schemed group configuration is synced with the server. Since the name isn't user-visible (but required for validation), I just went with 'Group Configuration', but I can change it to 'Content Group Configuration' to match what's actually returned by the server.